src/java.base/share/classes/java/io/SimpleIO.java

```diff
@@ -0,0 +1,1031 @@
+                Module lrbModule = lrbClass.getModule();
+                Module baseModule = Object.class.getModule();
+                baseModule.addReads(lrbModule);
+                MethodHandles.Lookup lookup = MethodHandles.lookup();
+                MethodHandle builderMH = lookup.findStatic(lrbClass, "builder", MethodType.methodType(lrbClass));
+                MethodHandle buildMH = lookup.findVirtual(lrbClass, "build", MethodType.methodType(lrClass));
+                readLineMH = lookup.findVirtual(lrClass, "readLine",
+                        MethodType.methodType(String.class, String.class));
+                readLineMaskMH = lookup.findVirtual(lrClass, "readLine",
+                        MethodType.methodType(String.class, String.class, Character.class));
+                Object builder = builderMH.invoke();
+                lineReader = buildMH.invoke(builder);
+            } catch (Throwable ex) {
+                readLineMH = null;
+                readLineMaskMH = null;
+                lineReader = null;
+            }
+
+            READ_LINE_MH = readLineMH;
+            READ_LINE_MASK_MH = readLineMaskMH;
+            LINE_READER = lineReader;
+        }
+
+        /**
+         * {@return true if LineReader is available.}
+         */
+        private static boolean hasLineReader() {
+            return LINE_READER != null;
+        }
+
+        /**
+         * Invoke LineReader::readLine.
+         *
+         * @param prompt Read line prompt.
+         * @return Line read in.
+         */
+        private static String readLine(String prompt) {
+            try {
+                return (String) READ_LINE_MH.invoke(LINE_READER, prompt);
+            } catch (Throwable ex) {
+                return null;
+            }
+        }
+
+        /**
+         * Invoke LineReader::readLine with mask.
+         *
+         * @param prompt Read line prompt.
+         * @return Line read in.
+         */
+        private static String readLine(String prompt, Character mask) {
+            try {
+                return (String) READ_LINE_MASK_MH.invoke(LINE_READER, prompt, mask);
+            } catch (Throwable ex) {
+                return null;
+            }
+        }
+    }
+
+    /**
+     * Return a string of characters from input after issuing a prompt. Unlike other
+     * input methods, this method supports editing and navigation of the input, as well
+     * as scrolling back through historic input. For example:
+     * {@snippet lang="java":
+     * var name = input("Name>> "); // @highlight substring="input"
+     * printLine(name);
+     * }
+     * will interact with the console as:
+     * {@snippet lang="text":
+     * Name>> Jean
+     * Jean
+     * }
+     *
+     * @param prompt string contain prompt for input, may be the empty string
+     * @return a string of characters read in from input or empty string if user aborts input
+     * @throws NullPointerException if prompt is null
```

Comment:
Throwing `NullPointerException` is indicated only here and in `printLines`, but I think it would be thrown in numerous other places, right? If so, then perhaps add a statement to that effect to the class level specification and remove the various `@throws NullPointerException`. Either that or add the latter everywhere it is needed.